src/domain/search/utils.js

```diff
@@ -51,8 +51,11 @@ export const getApiParamsFromFilters = (filters) => {
   return params;
 };
 
-export const getUnitOptions = (units, locale) => {
-  const options = units.map((unit) => {
+export const getUnitOptions = (units, locale, municipality) => {
+  const unitsFilteredByMunicipality = (municipality
```

Comment:
I know this is react style JavaScript code, but I feel this takes an unnecessarily long amount of time to actually understand what is happening here.

I would refactor this in this way:
```js
let unitsFilteredByMunicipality = units;

if (municipality) {
    const municipalities = municipality.split(',');
    unitsFilteredByMunicipality = units.filter(unit => municipalities.includes(unit.municipality));
}

const options = unitsFilteredByMunicipality.map(unit => {
...
```

Each line above does one thing and understanding it should be considerably easier than fitting almost all functionality on one line. What do you think?

Reply:
I agree that it's more readable when spiltted to multiple lines. Made the suggested changes now